index.js

```diff
@@ -41,7 +41,11 @@ module.exports = (urlString, opts) => {
 
 	// Remove duplicate slashes
 	if (urlObj.pathname) {
-		urlObj.pathname = urlObj.pathname.replace(/\/{2,}/g, '/');
+		if (/https?:/.test(urlObj.pathname)) {
+			urlObj.pathname = urlObj.pathname.replace(/\/{3,}/g, '/');
```

Comment:
I don't think this is optimal. You would still want `https://foo.com/https://bar.com/foo//bar` to be deduplicated. Would be better to make the regex deduplicate everything except the start if it starts with `https://` or `http://`.

Reply:
First of all, thanks for the reply. 

So if I understand it correctly, we want to remove second `/` from `foo//bar` in `https://foo.com/https://bar.com/foo//bar` right? 

If thats the case i have it this way:
```javascript
// Remove duplicate slashes
if (urlObj.pathname) {
  urlObj.pathname = urlObj.pathname.replace(/\/{2,}/g, '/');
  urlObj.pathname = urlObj.pathname.replace(/(http|https):\//, '$1://');
}
```

I remove all the double `/` ( which was the case) and then explicitly check for `http:/` and `https:/` and replace it  with `http://` and `https://`. Let me know if this approach is okay or not.

Reply:
> urlObj.pathname = urlObj.pathname.replace(/(http|https):\//, '$1://');

This is not a safe replacement. You're not making sure it's the start of the string. And it would also mean `https://foo.com/https:/bar.com` would be incorrectly normalized to `https://foo.com/https://bar.com`.

Reply:
We can't assert it at the start of the string because usually there are options in the url before actual url.

For example let's take this url, `https://images.thestar.com/xQ9RYsUCZcrg7zFSrY7pUybYqmQ=/1200x800/smart/filters:cb(1529514000000)/https://www.thestar.com/content/dam/thestar/news/world/2018/06/19/former-trump-staffer-corey-lewandowski-dismisses-story-of-immigrant-10-year-old-with-down-syndrome-taken-from-parents-with-wah-wah/lewandowski.jpg` on [this](https://www.thestar.com/news/world/2018/06/19/former-trump-staffer-corey-lewandowski-dismisses-story-of-immigrant-10-year-old-with-down-syndrome-taken-from-parents-with-wah-wah.html) page.

`urlObj.pathname` for the above url has `xQ9RYsUCZcrg7zFSrY7pUybYqmQ=/1200x800/smart/filters:cb(1529514000000)/https://www.thestar.com/content/dam/thestar/news/world/2018/06/19/former-trump-staffer-corey-lewandowski-dismisses-story-of-immigrant-10-year-old-with-down-syndrome-taken-from-parents-with-wah-wah/lewandowski.jpg`. 

Unless you want to split on `/` and then check if the string in the array of strings starts with `https:/` and then replace that? 

Even [this libray](https://github.com/willnorris/imageproxy) (1000+ stars) has the form `http://localhost/{options}/{remote_url}` and thus `urlObj.pathname` will not start with a `https:/` protocol.


>And it would also mean https://foo.com/https:/bar.com would be incorrectly normalized to https://foo.com/https://bar.com.

Also, is `https://foo.com/https:/bar.com` even a valid url? 

Happy to make changes that you feel comfortable with.

Reply:
> Even this libray (1000+ stars) has the form http://localhost/{options}/{remote_url} and thus urlObj.pathname will not start with a https:/ protocol.

Alright. I didn't consider that case.

> Also, is https://foo.com/https:/bar.com even a valid url?

Why wouldn't it when `https://foo.com/https://bar.com` is valid.

I think the best way to handle this is to only deduplicate `/` when they're not prefixed by `https?` in one step.

Reply:
>Why wouldn't it when https://foo.com/https://bar.com is valid.

Yes, you are right. 

>I think the best way to handle this is to only deduplicate `/` when they're not prefixed by `https?` in one step.

I have updated the regex to remove duplicate slashes to this.

```javascript
// Remove duplicate slashes
if (urlObj.pathname) {
  urlObj.pathname = urlObj.pathname.replace(/(?<!https?:)\/{2,}/g, '/');
}
```

Happy to commit and merge if you think this works. :)

Reply:
That's a good regex for the future, but lookbehind regex syntax is only available in Node.js 10, so we can't use it yet (We can put it in a comment for future use when we target Node.js 10 though).

Reply:
Good catch. I have updated the regex and here is the code below.

```javascript
if (urlObj.pathname) {
  urlObj.pathname = urlObj.pathname.replace(/((?![https?:]).)\/{2,}/g, (_, p1) => {
    if (/^(?!\/)/g.test(p1)) {
      return `${p1}/`;
    }
    return '/';
  });
}
```

This regex does capture a group if `//` aren't preceded with `http?:` so i have to manually check if the precedding character is `/` or something else. 

I know it is kind of ugly, but once we target only versions greater than node 9.11, we can switch over to cleaner regex.